packages/beacon-node/src/monitoring/system.ts

```diff
@@ -0,0 +1,268 @@
+  // physicalCores in some cases is the count of logical CPU cores
+  cpuCores = cpu.physicalCores;
+  cpuThreads = cpu.cores;
+
+  miscOs = getNormalizedOsVersion();
+
+  staticDataCollected = true;
+}
+
+async function collectCpuData(): Promise<void> {
+  const cpuTimes: Record<string, number> = {};
+
+  os.cpus().forEach((cpu) => {
+    // sum up CPU times per mode and convert to seconds
+    for (const [mode, time] of Object.entries(cpu.times)) {
+      if (cpuTimes[mode] == null) cpuTimes[mode] = 0;
+      cpuTimes[mode] += Math.floor(time / 1000);
+    }
+  });
+
+  // Note: currently beaconcha.in expects system CPU seconds to be everything
+  cpuNodeSystemSecondsTotal = Object.values(cpuTimes).reduce((total, time) => total + time, 0);
+  cpuNodeUserSecondsTotal = cpuTimes.user;
+  cpuNodeIdleSecondsTotal = cpuTimes.idle;
+}
+
+async function collectMemoryData(): Promise<void> {
+  const memory = await system.mem();
+  memoryNodeBytesTotal = memory.total;
+  memoryNodeBytesFree = memory.free;
+  memoryNodeBytesCached = memory.cached;
+  memoryNodeBytesBuffers = memory.buffers;
+}
+
+async function collectDiskData(): Promise<void> {
+  const fileSystems = await system.fsSize();
+  // get file system root, on windows this is the name of the hard disk partition
+  const rootFs = process.platform === "win32" ? process.cwd().split(path.sep)[0] : "/";
+  // only consider root file system, if it does not exist use first entry in the list
+  const fileSystem = fileSystems.find((fs) => fs.mount === rootFs) ?? fileSystems[0];
+  diskNodeBytesTotal = fileSystem.size;
+  diskNodeBytesFree = fileSystem.available;
+
+  if (diskIOMeasurable) {
+    const disk = await system.disksIO();
+    if (disk != null && disk.rIO !== 0) {
+      // Note: rIO and wIO might not be available inside container
+      // see https://github.com/sebhildebrandt/systeminformation/issues/777
+      diskNodeReadsTotal = disk.rIO;
+      diskNodeWritesTotal = disk.wIO;
+    } else {
+      diskIOMeasurable = false;
+    }
+  }
+}
+
+async function collectNetworkData(): Promise<void> {
+  // defaults to first external network interface
+  const [network] = await system.networkStats();
+  // Note: rx_bytes and tx_bytes will be inaccurate if process
+  // runs inside container as it only captures local network traffic
+  networkNodeBytesTotalReceive = network.rx_bytes;
+  networkNodeBytesTotalTransmit = network.tx_bytes;
+}
+
+function getNormalizedOsVersion(): MiscOs {
+  switch (process.platform) {
+    case "linux":
+      return "lin";
+    case "darwin":
+      return "mac";
+    case "win32":
+      return "win";
+    default:
+      return "unk";
+  }
+}
+
+function getSystemBootTime(): number {
+  return Math.floor(Date.now() / 1000 - os.uptime());
+}
+
+/**
+ * Number of CPU cores available
+ */
+export function getCpuCores(): number {
+  return cpuCores;
+}
+
+/**
+ * Number of CPU threads available
+ */
+export function getCpuThreads(): number {
+  return cpuThreads;
+}
```

Comment:
Overall seems good to me. I don't prefer to have `get` prefix for so many functions. Instead an object with getter properties or a class with getters would be preferred.

Reply:
I wanted to use a class initially but then I was no longer sure it makes due to the way the values should be cached. Should I define it as a class, instantiate it in this file and export the instantiated class only from the file? I could see that, should be the same result but everything wrapped into a class

Reply:
Refactored in https://github.com/ChainSafe/lodestar/pull/5202